metadata-ingestion/tests/unit/report_generator/test_report_generator.py

```diff
@@ -0,0 +1,218 @@
+import os
+import tempfile
+from unittest import TestCase
+from unittest.mock import MagicMock, patch
+
+import requests
+
+import datahub.cli.generate_report.report_generator
+from datahub.cli.generate_report.report_generator import (
+    PrivacyTermExtractor,
+    ReportGenerator,
+)
+
+FIXTURES_PATH = os.path.dirname(__file__)
+
+class TestReportGenerator(TestCase):
+
+    def setUp(self):
+        self.tmp_output_file = tempfile.NamedTemporaryFile()
+        self.config = {
+            "datahub_base_url": "http://localhost:1234",
+            "search_queries": ['*'],
+            "output": {
+                "format": "csv",
+                "destination": {
+                    "filepath": self.tmp_output_file.name
+                },
+            },
+        }
+
+    def tearDown(self):
+        self.tmp_output_file.close()
+
+    @patch.object(datahub.cli.generate_report.report_generator, 'PrivacyTermExtractor')
+    def test_basic(self, mock_privacy_term_extractor_class):
+        mock_extractor = MagicMock()
+        mock_extractor.yield_search_results.return_value = [
+            {
+                'dataset': 'stage_db.users.user',
+                'field': 'id',
+                'type': ['IDENTIFIER'],
+                'privacy_law': ['CCPA', 'GDPR'],
+            },
+            {
+                'dataset': 'stage_db.users.user',
+                'field': 'full_name',
+                'type': ['PERSON'],
+                'privacy_law': ['CCPA', 'GDPR'],
+            },
+            {
+                'dataset': 'stage_db.users.user',
+                'field': 'created',
+                'type': ['DATE'],
+                'privacy_law': [],
+            },
+            {
+                'dataset': 'stage_db.users.user',
+                'field': 'updated',
+                'type': ['DATE'],
+                'privacy_law': [],
+            },
+            {
+                'dataset': 'stage_db.users.user',
+                'field': 'field_without_terms',
+                'type': [],
+                'privacy_law': [],
+            },
+        ]
+        mock_privacy_term_extractor_class.return_value = mock_extractor
+
+        rg = ReportGenerator.create(self.config)
+        rg.generate()
+
+        mock_privacy_term_extractor_class.assert_called_once_with(self.config['datahub_base_url'])
+        mock_extractor.yield_search_results_assert_called_once_with(self.config['search_queries'])
+
+        with open(os.path.join(FIXTURES_PATH, 'expected_basic.csv'), 'rb') as f:
+            expected_lines = f.readlines()
+
+        self.tmp_output_file.seek(0)
+        actual_lines = self.tmp_output_file.readlines()
+        for expected, actual in zip(expected_lines, actual_lines):
+            self.assertEqual(expected.strip(), actual.strip())
+
+class TestPrivacyTermExtractor(TestCase):
```

Comment:
Could we add a test including pagination?

Reply:
good call

Reply:
done https://github.com/Affirm/datahub/pull/3/commits/3deac44ed9124d35c1364a60d1e1a94ea75de193#diff-1c8395edf9c226c152248c8fe6349692f13a4eb4201eb9c3c05c5e49dcbd4fdbR212-R282